lib/Models/registerCatalogMembers.ts

```diff
@@ -55,4 +56,77 @@ export default function registerCatalogMembers() {
   );
   CatalogMemberFactory.register(MagdaReference.type, MagdaReference);
   CatalogMemberFactory.register(KmlCatalogItem.type, KmlCatalogItem);
+
+  createCatalogItemFromUrl.register(
+    matchesExtension("csv"),
+    CsvCatalogItem.type
+  );
+  createCatalogItemFromUrl.register(
+    matchesExtension("czm"),
+    CzmlCatalogItem.type
+  );
+  createCatalogItemFromUrl.register(
+    matchesExtension("czml"),
+    CzmlCatalogItem.type
+  );
+  createCatalogItemFromUrl.register(
+    matchesExtension("geojson"),
+    GeoJsonCatalogItem.type
+  );
+  createCatalogItemFromUrl.register(
+    matchesExtension("json"),
+    GeoJsonCatalogItem.type
+  );
+  createCatalogItemFromUrl.register(
+    matchesExtension("kml"),
+    KmlCatalogItem.type
+  );
+  createCatalogItemFromUrl.register(
+    matchesExtension("kmz"),
+    KmlCatalogItem.type
+  );
+  createCatalogItemFromUrl.register(
+    matchesExtension("topojson"),
+    GeoJsonCatalogItem.type
+  );
+
+  // These items work by trying to match a URL, then loading the data. If it fails, they move on.
+  createCatalogItemFromUrl.register(
+    matchesUrl(/\/wms/i),
+    WebMapServiceCatalogGroup.type,
+    true
+  );
+  createCatalogItemFromUrl.register(
+    matchesUrl(/\/arcgis\/rest\/.*\/MapServer\/\d+\b/i),
```

Comment:
@steve9164 @rowanwins sorry need your help here - I'm not familiar enough to know if all mapservers have arcgis or rest preceeding it? It's not enforced, but at least some sort of default?

Reply:
I can't think of any Esri/ArcGIS map services that don't have `/MapServer`.

We'd want to auto-detect `/MapServer`(which is a group of layers) and also `/MapSever/0` (which is a specific layer).